ansible/roles/repo/tasks/redhat.yaml

```diff
@@ -12,6 +12,20 @@
     - lookup('env', 'RHSM_PASS') | length > 0
     - ansible_distribution == 'RedHat'
 
+- name: create proxy file
+  file:
```

Comment:
Were you able to repro this issue, see `default_env` [already has this defined](https://github.com/mesosphere/konvoy-image-builder/blob/3080f07e534aa4b470f9a024580b191a88474d61/ansible/group_vars/all/system.yaml#L37) and then it gets set in [provision.yaml](https://github.com/mesosphere/konvoy-image-
builder/blob/3080f07e534aa4b470f9a024580b191a88474d61/ansible/provision.yaml#L43).

We did something [similar to that in Konvoy1.x ](https://github.com/mesosphere/konvoy/blob/a225484be0b7612d5d00d305adf5bc539bf1d130/ansible/group_vars/all/system.yaml#L210) and didn't face this issue.

My concern is that this is only solving it for `yum` but we also have `apt`, `package` and `get_url` tasks that also need this.

Is it possible that this issue exists because of the Ansible version we are using here and its solved later? [yum It should already be reading it from the environment](https://github.com/sillent/ansible/blob/3f0900f504f911dcb4a61e1520a2a6e9967a5a27/lib/ansible/modules/packaging/os/yum.py#L713).

Reply:
Yes I was able to reproduce it and it seems like yum doesn’t respect the http_proxy setting

Reply:
```
TASK [repo : RHEL subscription] ************************************************
skipping: [35.93.107.242]

TASK [repo : add alma linux extra repository] **********************************
ok: [35.93.107.242]

TASK [repo : inspect env vars] *************************************************
ok: [35.93.107.242] => {
    "ansible_facts.env": {
        "HOME": "/root",
        "HTTPS_PROXY": "http://54.212.37.133:8888",
        "HTTP_PROXY": "http://54.212.37.133:8888",
        "LANG": "en_US.UTF-8",
        "LOGNAME": "root",
        "LS_COLORS": "rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=01;05;37;41:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=01;36:*.au=01;36:*.flac=01;36:*.m4a=01;36:*.mid=01;36:*.midi=01;36:*.mka=01;36:*.mp3=01;36:*.mpc=01;36:*.ogg=01;36:*.ra=01;36:*.wav=01;36:*.oga=01;36:*.opus=01;36:*.spx=01;36:*.xspf=01;36:",
        "MAIL": "/var/mail/root",
        "NO_PROXY": "",
        "PATH": "/sbin:/bin:/usr/sbin:/usr/bin",
        "PWD": "/home/ec2-user",
        "PYTHONPATH": "",
        "SHELL": "/bin/bash",
        "SHLVL": "1",
        "SUDO_COMMAND": "/bin/sh -c echo BECOME-SUCCESS-bxamkbyxarqizdbvglolcdzuoajjyiad ; HTTPS_PROXY=http://54.212.37.133:8888 https_proxy=http://54.212.37.133:8888 HTTP_PROXY=http://54.212.37.133:8888 http_proxy=http://54.212.37.133:8888 NO_PROXY='' no_proxy='' PYTHONPATH='' /usr/libexec/platform-python /home/ec2-user/.ansible/tmp/ansible-tmp-1674612490.734585-45-280473584369634/AnsiballZ_setup.py",
        "SUDO_GID": "1000",
        "SUDO_UID": "1000",
        "SUDO_USER": "ec2-user",
        "TERM": "xterm",
        "USER": "root",
        "_": "/usr/libexec/platform-python",
        "http_proxy": "http://54.212.37.133:8888",
        "https_proxy": "http://54.212.37.133:8888",
        "no_proxy": ""
    }
}

TASK [repo : install epel-release for RHEL 8.x] ********************************
fatal: [35.93.107.242]: FAILED! => {"changed": false, "msg": "Failed to download metadata for repo 'alma-linux-extra': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried", "rc": 1, "results": []}
```

when reproducing this i created a instance that has a security group that can only reach my proxy and i was able to repro

Reply:
when building with my branch 

```
09:53 PM  faiqraza @ linux  ~/go/src/github.com/mesosphere/konvoy-image-builder  faiq/D2IQ-95440⚡? 
$ ANSIBLE_STRATEGY=debug ANSIBLE_ENABLE_TASK_DEBUGGER=True ./bin/konvoy-image-wrapper provision --inventory-file inventory.yaml --overrides=overrides/http_proxy.yaml 
\\2023/01/25 02:53:25 writing new packer configuration to work/provision_build-1674615205-ikvuY
[DEPRECATION WARNING]: "include" is deprecated, use include_tasks/import_tasks 
instead. This feature will be removed in version 2.16. Deprecation warnings can
 be disabled by setting deprecation_warnings=False in ansible.cfg.

PLAY [Prepare Ansible - Flatcar Python setup] **********************************

TASK [Flatcar no update] *******************************************************
ok: [35.93.107.242]

TASK [exec bootstrap python] ***************************************************
ok: [35.93.107.242]

TASK [check for flatcar python] ************************************************
ok: [35.93.107.242]

TASK [change python if bootstrapped] *******************************************
skipping: [35.93.107.242]

PLAY [Provision] ***************************************************************

TASK [Gathering Facts] *********************************************************
ok: [35.93.107.242]

TASK [include_vars] ************************************************************
skipping: [35.93.107.242]

TASK [offline : include_tasks] *************************************************
included: /tmp/kib/ansible/roles/offline/tasks/upload.yaml for 35.93.107.242

TASK [offline : Gather os-release facts] ***************************************
ok: [35.93.107.242]

TASK [offline : Set containerd_tar_file] ***************************************
ok: [35.93.107.242]

TASK [offline : create images directory] ***************************************
skipping: [35.93.107.242]

TASK [offline : list tar.gz files in the local image cache] ********************
skipping: [35.93.107.242]

TASK [offline : upload image bundles to remote] ********************************
skipping: [35.93.107.242]

TASK [offline : upload mindthegap] *********************************************
skipping: [35.93.107.242]

TASK [offline : create offline OS packages directory] **************************
skipping: [35.93.107.242]

TASK [offline : upload OS packages bundle to remote] ***************************
skipping: [35.93.107.242]

TASK [offline : remove outdated os packages] ***********************************
skipping: [35.93.107.242]

TASK [offline : create offline OS packages directory after cleanup] ************
skipping: [35.93.107.242]

TASK [offline : unarchive OS packages bundle on remote] ************************
skipping: [35.93.107.242]

TASK [offline : create offline pip packages directory] *************************
skipping: [35.93.107.242]

TASK [offline : upload pip packages bundle to remote] **************************
skipping: [35.93.107.242]

TASK [offline : remove outdated pip packages] **********************************
skipping: [35.93.107.242]

TASK [offline : create offline pip packages directory after cleanup] ***********
skipping: [35.93.107.242]

TASK [offline : unarchive pip packages bundle on remote] ***********************
skipping: [35.93.107.242]

TASK [offline : create offline containerd bundle directory] ********************
skipping: [35.93.107.242]

TASK [offline : upload containerd bundle to remote] ****************************
skipping: [35.93.107.242]

TASK [offline : create nvidia directory] ***************************************
skipping: [35.93.107.242]

TASK [offline : upload runfile artifact] ***************************************
skipping: [35.93.107.242]

TASK [offline : check for offline bundle] **************************************
ok: [35.93.107.242]

TASK [offline : set offline enabled fact] **************************************
ok: [35.93.107.242]

TASK [repo : RHEL subscription] ************************************************
skipping: [35.93.107.242]

TASK [repo : create proxy file] ************************************************
changed: [35.93.107.242]

TASK [repo : modify proxy settings if http_proxy is present] *******************
ok: [35.93.107.242]

TASK [repo : inspect env vars] *************************************************
ok: [35.93.107.242] => {
    "ansible_facts.env": {
        "HOME": "/root",
        "HTTPS_PROXY": "http://54.212.37.133:8888",
        "HTTP_PROXY": "http://54.212.37.133:8888",
        "LANG": "en_US.UTF-8",
        "LOGNAME": "root",
        "LS_COLORS": "rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=01;05;37;41:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=01;36:*.au=01;36:*.flac=01;36:*.m4a=01;36:*.mid=01;36:*.midi=01;36:*.mka=01;36:*.mp3=01;36:*.mpc=01;36:*.ogg=01;36:*.ra=01;36:*.wav=01;36:*.oga=01;36:*.opus=01;36:*.spx=01;36:*.xspf=01;36:",
        "MAIL": "/var/mail/root",
        "NO_PROXY": "",
        "PATH": "/sbin:/bin:/usr/sbin:/usr/bin",
        "PWD": "/home/ec2-user",
        "PYTHONPATH": "",
        "SHELL": "/bin/bash",
        "SHLVL": "1",
        "SUDO_COMMAND": "/bin/sh -c echo BECOME-SUCCESS-hfkinyvzuysrnqxzhdngzcwzodqnyyau ; HTTPS_PROXY=http://54.212.37.133:8888 https_proxy=http://54.212.37.133:8888 HTTP_PROXY=http://54.212.37.133:8888 http_proxy=http://54.212.37.133:8888 NO_PROXY='' no_proxy='' PYTHONPATH='' /usr/libexec/platform-python /home/ec2-user/.ansible/tmp/ansible-tmp-1674615221.0384045-45-234752720022461/AnsiballZ_setup.py",
        "SUDO_GID": "1000",
        "SUDO_UID": "1000",
        "SUDO_USER": "ec2-user",
        "TERM": "xterm",
        "USER": "root",
        "_": "/usr/libexec/platform-python",
        "http_proxy": "http://54.212.37.133:8888",
        "https_proxy": "http://54.212.37.133:8888",
        "no_proxy": ""
    }
}

TASK [repo : add alma linux extra repository] **********************************
ok: [35.93.107.242]
```

Reply:
Interesting, so this may be an issue with the `yum_repository` module, or does it fail with other `yum` tasks?

Can you please try setting the `proxy` [paramtere](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yum_repository_module.html#parameter-proxy), I really would like to avoid making host level changes.

Reply:
Actually they may actually write the proxy into the `repo` file, another option could be to set `envrinoment` arg on the task and see if that helps.

Reply:
let me try that as well. 

i did it on the host level so that other yum stuff could also pass, but lets try this way as well